### PR TITLE
The Project is for To Do list: interactive list

### PR DESCRIPTION
**Pull request by SAZZAD1Q2**
Added a new JavaScript file and import it as a module:
Added event listener to the checkbox.
Updated items object's value for completed key upon user actions.

### CODE REVIEW

File: src/modules/edit.js [hunk 1]
```diff
@@ -1,39 +1,11 @@
...
-    text: taskValue,
-    completed: false,
-    index: savedTasks.length + 1,
-  };
-
-  savedTasks.push(task);
+export function updateStatus(index, completed) {
+  savedTasks[index].completed = completed;
   saveTasks();
 }
```
Comment (TadesseDev) [inline src/modules/edit.js]: - [ ] dear @SAZZAD1Q2, Here for the completed methods we need to have a new JavaScript file imported as a module. This is to make sure you understand importing modules in your project. Please implement the functionality for status updates (`completed`: `true` / `false`) in a new file and import it as a module. 
- [ ] Also, when the user clicks on the check box, we need to make sure the task status is changed, Also the change should be stored in the local storage. This makes your app a lot more flexible.

File: src/index.js [hunk 2]
```diff
@@ -131,6 +139,14 @@ function addTask() {
   updateIndexes();
 }
 
-renderTasks();
+function clearCompletedTasks() {
+  const completedTasks = savedTasks.filter((task) => task.completed);
+  completedTasks.forEach((task) => {
+    const index = savedTasks.indexOf(task);
+    removeTask(index);
+  });
```
Comment (TadesseDev) [inline src/index.js]: - [ ] Also here, Your clear all completed button is expected to clear all the completed tasks from the local storage, In your case, it just unchecks the check box and this is not the expected behavior. Please make sure all the completed tasks are removed totally from the local storage on the button clicked.
- [ ] Also, there is a spelling error in your `clear all completed button`. Please rename `Clear as Complete` to `Clear all Completed`

File: src/modules/edit.js [hunk 3]
```diff
@@ -1,39 +1,11 @@
-export const savedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+import { saveTasks, savedTasks } from './localStorage.js';
 
-export function saveTasks() {
```
Comment (TadesseDev) [inline src/modules/edit.js]: **Dear @SAZZAD1Q2, at this point, the user is expected to be able to mark task completion by selecting the corresponding checkbox. The updated tasks list should be stored in local storage. All the elements of the user interface should be fully functional and the application should be completed at this stage.**
> In your case 
- [ ] Your task is not loaded from the local storage when the app starts. 
- [ ] checkbox changes are not getting saved to the local storage. 
- [ ] `clear all completed` button is not working 
**Please fix this issues before the next review request**

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @SAZZAD1Q2 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitHub Flow  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — CHANGES_REQUESTED** [review]
Hi @SAZZAD1Q2 ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/index.js [hunk 4]
```diff
@@ -64,6 +70,8 @@ function editTask(li) {
 }
 
 function renderTasks() {
+  todoList.innerHTML = ''; // Clear the existing list before rendering
+
   savedTasks.forEach((task, index) => {
```
Comment (shella12) [inline src/index.js:67]: - [ ] Great job on saving/loading the list to/from the localStorage however as suggested by the previous reviewer please make sure that when some boxes are marked completed that particular task status also updates in the local storage and when the page is reloaded the tasks that were marked completed stay marked completed

File: src/index.js [hunk 5]
```diff
@@ -4,6 +4,15 @@ import { saveTasks, savedTasks } from './modules/edit.js';
...
 const todoList = document.getElementById('todo-list');
+const clearButton = document.getElementById('clearButton');
+
+// Load tasks from local storage
+function loadTasks() {
+  const tasks = localStorage.getItem('tasks');
+  if (tasks) {
+    savedTasks.push(...JSON.parse(tasks));
+  }
+}
```
Comment (shella12) [inline src/index.js]: - [ ] Upon reloading the page, the list of todo list is just multiplying please make sure your list is only populated with the number of tasks that were already present in the list once

https://github.com/SAZZAD1Q2/New-TO-Do-List/assets/44798044/0688cbd4-1ecd-415a-8756-c722225b1d8b

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @SAZZAD1Q2, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. You are almost there...



## Highlights

- Clear all Completed function is working ✅ 
- Data is populated from localStorage ✅   

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by souad988 — APPROVED** [review]
Hi @SAZZAD1Q2 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Good file structure
Use webpack correctly
Create modules to separate functionalities

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**